Source/WebCore/html/parser/HTMLEntitySearch.cpp

```diff
@@ -45,9 +46,10 @@ HTMLEntitySearch::HTMLEntitySearch()
 
 HTMLEntitySearch::CompareResult HTMLEntitySearch::compare(const HTMLEntityTableEntry* entry, UChar nextCharacter) const
 {
+    const LChar* entityString = HTMLEntityTable::characters(*entry);
+    UChar entryNextCharacter = entityString[m_currentLength];
```

Comment:
No, this is not OK. Can’t do this before checking `m_currentLength` agains `entry->length`.

Reply:
Sure, I will do it. :-)

Reply:
Just moved below to original place after `if (entry->length < m_currentLength...` check. Will push next commit, after confirming if you have any other comments or if I have addressed them incorrectly. Thanks!

Reply:
Done. :-)